src/pages/index.js

```diff
@@ -60,66 +97,146 @@ const editCardPopup = new PopupWithForm(
 );
 editCardPopup.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithForm("#delete-card-modal", () => {});
+deleteCardPopup.setEventListeners();
+
 // Import from PopupWithImage
 
 const imagePopUp = new PopupWithImage("#preview-image-modal");
 imagePopUp.setEventListeners();
 
-// Profile DOM nodes
-const profileTitle = document.querySelector(".profile__title");
-const profileDescription = document.querySelector(".profile__description");
-
 // Import from UserInfo
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 // Buttons and other DOM nodes
 const profileEditButton = document.querySelector(".profile__edit-button");
 const addNewCardButton = document.querySelector(".profile__add-button");
-const profileEditCloseButton = profileEditModal.querySelector(".modal__close");
-const addCardModalCloseButton = addCardModal.querySelector(".modal__close");
-const previewImageCloseButton =
-  previewImageModal.querySelector(".modal__close");
+const editAvatarButton = document.querySelector(".profile__image");
+const editAvatarButtonPencil = document.querySelector(".profile__pencil");
 
 // FORM DATA
 const profileTitleInput = profileEditForm.querySelector(".modal__input_name");
 const profileDescriptionInput = profileEditForm.querySelector(
   ".modal__input_description"
 );
-const cardTitleInput = addCardFormElement.querySelector(".modal__input_title");
-const cardLinkInput = addCardFormElement.querySelector(".modal__input_link");
 
 // FUNCTIONS
 
+function handleAvatarEditSubmit(e, formValues) {
+  e.preventDefault();
+
+  const { avatar } = formValues;
+  console.log("Avatar URL being sent:", avatar);
+
+  editAvatarPopup.setLoadingState(false);
+
+  api
+    .updateUserAvatar(avatar)
+    .then(() => {
+      return api.getUserInfo();
```

Comment:
` api.getUserInfo();` is not needed when you edit the profile data or change the `avatar` because you have the whole profile data in the **response** from the server